### PR TITLE
Univ declarations can have extensible qualities (internally)

### LINKED ISSUE #19072: abstract tactic doesn't work when the top-level proof is sort-polymorphic
### Description of the problem

```coq
Polymorphic Definition demo@{srt| |} : True.
Proof.
  abstract exact I.
  (* Quality srt is unbound. *)
```

### Version of Coq where this bug occurs

8.19.0

### Last version of Coq where the bug did not occur

_No response_

### PR DESCRIPTION
**Pull request by SkySkimmer**
Not supported in syntax as due to collapsing qualities any quality we
see must be named, but side effects use default_univ_decl which must
allow the named qualities from the surrounding proof.

Fix #19072

Overlays:
- https://github.com/LPCIC/coq-elpi/pull/632

### CODE REVIEW

**Comment by ppedrot** [comment]
@coqbot run full ci

File: engine/uState.ml [hunk 1]
```diff
@@ -826,8 +826,10 @@ let universe_context_inst decl qvars levels names =
   let leftus = List.fold_left (fun acc l -> Level.Set.remove l acc) levels decl.univdecl_instance in
   let () =
     let unboundus = if decl.univdecl_extensible_instance then Level.Set.empty else leftus in
-    if not (QVar.Set.is_empty leftqs && Level.Set.is_empty unboundus)
-    then error_unbound_universes leftqs unboundus names
+    (* allow unbound qualities, cf #19072
+       collapsing ensured that we only get named qualities anyway *)
+    if not (Level.Set.is_empty unboundus)
```
Comment (ppedrot) [inline engine/uState.ml:831]: This looks like looking for trouble if we want to generalize the use of sort polymorphic quantifications. We would get dangling sorts if we don't pay attention. Shouldn't we fix abstract instead?

Comment (SkySkimmer) [reply engine/uState.ml:831]: To change something on the abstract side we would need to introduce a notion of qvar extensible universe declaration
I guess we could but it's more work

Comment (ppedrot) [reply engine/uState.ml:831]: That'd be more in line with what we do with levels, wouldn't it? If so that may be the better way.

Comment (SkySkimmer) [reply engine/uState.ml]: changed

**Review by ppedrot — APPROVED** [review]

**Comment by proux01** [comment]
ping @ppedrot

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**